yt_dlp/extractor/sharepoint.py

```diff
@@ -0,0 +1,65 @@
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    float_or_none,
+    traverse_obj,
+)
+
+
+class SharePointIE(InfoExtractor):
+    _VALID_URL = r'https?://[^\.]+\.sharepoint\.com/:v:/g/[^/]+/[^/]+/(?P<id>[^/?]+)'
+
+    _TESTS = [
+        {
+            'url': 'https://lut-my.sharepoint.com/:v:/g/personal/juha_eerola_student_lab_fi/EUrAmrktb4ZMhUcY9J2PqMEBD_9x_l0DyYWVgAvp-TTOMw?e=ZpQOOw',
+            'info_dict': {
+                'id': 'EUrAmrktb4ZMhUcY9J2PqMEBD_9x_l0DyYWVgAvp-TTOMw',
+                'ext': 'mp4',
+                'title': 'CmvpJST.mp4',
+                'duration': 54.567,
+            },
+            'params': {'skip_download': 'm3u8'}
+        },
+    ]
+
+    def _parse_mediapackage(self, video, video_id):
+        formats = []
+
+        base_media_url = video['.transformUrl'].replace('/thumbnail?', '/videomanifest?')
+        base_media_url += f"&cTag={video['.ctag']}&action=Access&part=index"
```

Comment:
```suggestion
        base_media_url = (video['.transformUrl'].replace('/thumbnail?', '/videomanifest?')
                          + '&cTag={video[".ctag"]}&action=Access&part=index'))
```
Normally, I dislike +, but this is a good exception since below looks ugly
```py
        base_media_url = (f'{video[".transformUrl"].replace("/thumbnail?", "/videomanifest?")}'
                          f'&cTag={video[".ctag"]}&action=Access&part=index'))
```

Reply:
On the other hand, the proper way to modify the query does look better but is quite verbose...
```py
        parsed_url = urlparse(video['.transformUrl'])
        base_media_url = urlunparse(parsed_url._replace(
            path=urljoin(parsed_url.path, 'videomanifest'),
            query=urlencode({
                **parse_qs(parsed_url.query),
                'cTag': video['.ctag'],
                'action': 'Access',
                'part': 'index',
            }, doseq=True)))
```

Reply:
`urljoin(parsed_url.path, 'videomanifest')` is wrong here

Reply:
I guess that depends on the definition of wrong:
```py
>>> from urllib.parse import urljoin

>>> urljoin('https://www.example.com/something/thumbnail', 'videomanifest')
'https://www.example.com/something/videomanifest'

>>> urljoin('https://www.example.com/something/thumbnail/', 'videomanifest')
'https://www.example.com/something/thumbnail/videomanifest'

>>> urljoin('https://www.example.com/something/thumbnail/', '../videomanifest')
'https://www.example.com/something/videomanifest'
```
so it should be `urljoin(f'{parsed_url.path}/', '../videomanifest')` instead?

Reply:
just a side...  the echo360 source uses .replace("/thumbnail?", "/videomanifest?") ^^ thats where I copied it from. And they do just concatenate  the arguments depending on the options one by one. 

PS. I will probably work in all the suggestions tomorrow, today I have no time.

Reply:
I do not think the urlunparse construction is more readable but probably a tiny little bit more robust.